lib/utils.js

```diff
@@ -696,14 +696,17 @@ function* makeCounter(seed) {
 /**
  * Helper function for either accepting a callback, or returning a promise
  *
- * @param {Function} [callback] an optional callback.
+ * @param {Function|undefined} PromiseLibrary a promise library.
+ * @param {Function|undefined} callback an optional callback.
  * @param {Function} fn A function that takes a callback
  * @returns {Promise|void} Returns nothing if a callback is supplied, else returns a Promise.
  */
-function maybePromise(callback, fn) {
+function maybePromise(PromiseLibrary, callback, fn) {
```

Comment:
Originally, I was thinking of something like:
```
function maybePromise(parent, callback, fn) {
  const Promise = (parent.s && parent.s.promiseLibrary) || Promise;
  ...
```
since all of the things that `maybePromise` is called from follow that pattern. It seems like it would reduce the need to reference the `promiseLibrary` before each call, but maybe that's not a big deal. What do you think?

Reply:
* I thought of `maybePromise` as dumb / not knowing the implementation details for the parent class instance.
* However you're right it would trim down the code.

Reply:
implemented 👍